joomla/joomla-cms
# [4.0] Xtd modals done right

Pull Request for Issue #33210 .

### Summary of Changes
- Modals should not break when debug language is on
- Modals should have unique ids (since there is an option for multiple instances of an editor)

### Testing Instructions

- Enable debug and the language debug and also the display (the 2 switches that appear after enabling debug)
- Switch between Tinymce, Codemirror and none editor in the Global configuration
- Create an article and test ALL the xtd-buttons

### Actual result BEFORE applying this Pull Request

Broken

### Expected result AFTER applying this Pull Request

Fixed

### Documentation Changes Required

~~NO~~

~~Actually, yes! The PR https://github.com/joomla/joomla-cms/pull/19789 from 2018 introduced the property `realName` which basically is the untranslated string for the button and this is a B/C break and although buttons without it should still work the J4 implementation of the buttons **should** dictate for it for a11y and also to ensure proper functionality of the multiple instances of a browser. ~~

Rework the code so it should be B/C although J4 XTD Buttons should have a property `name` and another `icon` to ensure there won't be accidental conflicts
